### ISSUE TITLE
is performance slow down when timeline  is turned on?

### ISSUE THREAD

**Issue by tingweiwu** [horovod/horovod#627]
when I run 
 `mpirun -np 8 -H localhost:8 -x NCCL_DEBUG=ERROR python tensorflow_mnist.py `  

I got the  speed as follow 
```
INFO:tensorflow:loss = 0.020556288, step = 400 (0.101 sec)
INFO:tensorflow:global_step/sec: 98.331
INFO:tensorflow:loss = 0.07997544, step = 400 (0.102 sec)
```
but when I turn on timeline with the command 

`HOROVOD_TIMELINE=/log/timeline.json mpirun -x HOROVOD_TIMELINE -np 8 -H localhost:8 -x NCCL_DEBUG=ERROR python tensorflow_mnist.py`

I got the  speed as follow 
```
INFO:tensorflow:loss = 0.07853837, step = 400 (16.961 sec)
INFO:tensorflow:global_step/sec: 0.529734
INFO:tensorflow:loss = 0.0064633815, step = 400 (17.063 sec)
```

why the speed slow down so much when turn on timeline?

**Comment by alsrgv** [comment]
@tingweiwu, is `/log` on a local or shared filesystem?  If it's on a shared filesystem, it may have too slow I/O.

**Comment by lileilai** [comment]
@tingweiwu i am also try the tensorflow_mnist on my cluster, but my parameter setting as follows:

num_gpu=8    batch_size=64     log_interval_step=200
-----------------------------------------
But, i got the log as follows:
*********************************************************************
INFO:tensorflow:loss = 0.090276375, step = 200 (7.442 sec)
INFO:tensorflow:loss = 0.04766946, step = 200 (7.440 sec)
INFO:tensorflow:loss = 0.05376704, step = 200 (7.445 sec)
INFO:tensorflow:loss = 0.049182806, step = 200 (7.446 sec)
INFO:tensorflow:loss = 0.13704734, step = 200 (7.446 sec)
INFO:tensorflow:global_step/sec: 28.266
INFO:tensorflow:loss = 0.023688769, step = 200 (7.446 sec)
INFO:tensorflow:loss = 0.049734995, step = 200 (7.448 sec)
INFO:tensorflow:loss = 0.14634882, step = 200 (7.449 sec)
INFO:tensorflow:global_step/sec: 27.2771
INFO:tensorflow:loss = 0.042031713, step = 400 (7.483 sec)
INFO:tensorflow:loss = 0.02870454, step = 400 (7.479 sec)
INFO:tensorflow:loss = 0.02583262, step = 400 (7.487 sec)
INFO:tensorflow:global_step/sec: 26.2003
INFO:tensorflow:loss = 0.025815785, step = 400 (7.487 sec)
INFO:tensorflow:loss = 0.0047751926, step = 400 (7.482 sec)
INFO:tensorflow:loss = 0.0109404875, step = 400 (7.484 sec)
INFO:tensorflow:loss = 0.044682164, step = 400 (7.487 sec)
INFO:tensorflow:loss = 0.06950624, step = 400 (7.485 sec)
*************************************************************************
I got an low acceleration rate，could you have some suggestion or share your parameter config and code.

**Comment by tgaddair** [comment]
Hey @lileilai, `tensorflow_mnist.py` is not a good measure for scalability, as the model is too small to take advantage of data parallelism.  I would suggest trying `tensorflow_synthetic_benchmark.py` as a better baseline for scaling efficiency.

If you have further questions, feel free to open a new issue.  We'll keep this issue focused on the Horovod timeline feature.
